### ISSUE TITLE
Problem with the reconstruction loss

### ISSUE THREAD

**Issue by katjaschwarz** [zhaobozb/layout2im#5]
Hi Bo,

first of all thank you for providing the code to the paper - I found it really good to work with. However, running your training script code for the coco dataset (without changing the default values) I run into a problem. The reconstructed real images look just like the fake generated ones, but not like the real original images. 
I uploaded the tensorboard output here:
[Unbenanntes Dokument(1).pdf](https://github.com/zhaobozb/layout2im/files/3806027/Unbenanntes.Dokument.1.pdf)

I also tried to increase the reconstruction loss - which helped in the sense that real_rec and fake (random) do not look so similar anymore, but the reconstruction still does not work very well.
Do you have an idea what could be the problem here?

Best,
Katja

**Comment by katjaschwarz** [comment]
I also evaluated this for your pretrained model on coco and has the same problem.
Here are some of the outputs I get (the ordering is [rand, real, rec]): 
![img000000_rand](https://user-images.githubusercontent.com/32389516/68422282-2bd50f80-01a0-11ea-9a80-45d9aa1db85a.png)
![img000000_real](https://user-images.githubusercontent.com/32389516/68422283-2bd50f80-01a0-11ea-88dd-e50798c9f482.png)
![img000000_rec](https://user-images.githubusercontent.com/32389516/68422284-2bd50f80-01a0-11ea-9a6b-aecf52486e65.png)
![img000001_rand](https://user-images.githubusercontent.com/32389516/68422285-2bd50f80-01a0-11ea-9784-2cc912fe1d37.png)
![img000001_real](https://user-images.githubusercontent.com/32389516/68422286-2bd50f80-01a0-11ea-8505-a2cc60f6c884.png)
![img000001_rec](https://user-images.githubusercontent.com/32389516/68422287-2c6da600-01a0-11ea-8ff3-ddff53396cea.png)

Best,
Katja

**Comment by zhaobozb** [comment]
Hi Katja,

Thanks for your interests in our work. We also found that, using current hyper-parameters, the model cannot reconstruct the input image very well. Increasing the weight of reconstruction loss and/or decreasing the weight of KL divergence loss may help the model to better reconstruct the input image, but may also make generated images worse.

Cheers,
Bo

**Comment by katjaschwarz** [comment]
Okay, thanks!
